### ISSUE TITLE
ParseThat ls failing on Power

### ISSUE THREAD

**Issue by hainest** [dyninst/testsuite#47]
test_pt_ls,"",none,64,rewriter,NA,dynamic,nonPIC,FAILED

**Comment by hainest** [comment]
```
$ ./test_driver -64 -none -g++ -dynamiclink -create -nonpic -test test_pt_ls -log tmp.log
$ echo $?
2
$ cat tmp.log
Commencing test(s) ...
Sat Oct 19 16:28:52 PDT 2019
Linux ray23 3.10.0-862.14.4.1chaos.ch6.ppc64le #1 SMP Wed Sep 26 12:28:30 PDT 2018 ppc64le ppc64le ppc64le GNU/Linux
TESTDIR=/g/g20/haines8/paradyn/BkNZ0fM_/testsuite/tests
```

Running under gdb gives some insight

```
$ gdb --args ./test_driver -64 -none -gcc -dynamiclink -create -nonpic -test test_pt_ls -log tmp.log
(gdb) set env DYNINSTAPI_RT_LIB=/g/g20/haines8/paradyn/BkNZ0fM_/testsuite/dyninst/lib/libdyninstAPI_RT.so
(gdb) set env OMP_NUM_THREADS=2
(gdb) set env LD_LIBRARY_PATH=.:/g/g20/haines8/paradyn/BkNZ0fM_/dyninst/lib:/g/g20/haines8/paradyn/libraries/lib
(gdb) r
Starting program: /usr/WS1/haines8/BkNZ0fM_/testsuite/tests/./test_driver -64 -none -gcc -dynamiclink -create -nonpic -test test_pt_ls -log tmp.log
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Detaching after fork from child process 162970.
Detaching after fork from child process 162972.
[Inferior 1 (process 162964) exited with code 02]
Missing separate debuginfos, use: debuginfo-install libgcc-4.8.5-28.el7.ppc64le libstdc++-4.8.5-28.el7.ppc64le libxml2-2.9.1-6.el7_2.3.ppc64le xz-libs-5.2.2-1.el7.ppc64le zlib-1.2.7-17.el7.ppc64le
```

Why would separate debug info files be needed for this test?

**Comment by mxz297** [comment]
I don't think this test needs separate debug info files. Historically, this test is more likely to fail due to test set up issues, including paths for finding things. `-verbose` may give some useful information.

**Comment by hainest** [comment]
@mxz297 Sadly, the problem is with test_driver. For some reason, `test_pt_ls` is not being run on Power, so no tests are run at all (why this is so and doesn't return `SKIPPED` is yet another bug). The problem is that [`test_results_t::FAILED`](https://github.com/dyninst/testsuite/blob/master/src/test_results.h#L36) and [`NOTESTS`](https://github.com/dyninst/testsuite/blob/master/src/error.h#L37) have the same numerical value, so runTests gets confused.

**Comment by wrwilliams** [comment]
Yes, the missing separate debuginfos is a red herring here and pertains to your ability to debug into libstdc++/libgcc when debugging test_driver/parseThat.

The actual parseThat command for the test should be straightforward to extract and check manually. However, I'd also suggest that this test is trying to cover too many things with too small an example:

* Does parseThat build and run?
* Do we handle a stripped statically linked binary okay? (Note that this is not reliable for either of these cases, of course--but it incidentally tests both on many systems.)
* Do we handle a system binary (or more generally, non-local and not in a writeable location) okay?

If someone has time I would strongly recommend replacing this with more useful and controlled unit tests--a parseThat smoke test on something we build and control, and a couple of tests for simpler mutators working on "difficult" binaries.
